[SMART-79] Add PIN strength validation to "Define PIN" page

Anton-A-Sakovich opened the pull request.

Anton-A-Sakovich committed:
Add `EqualsParameter` value converter

Anton-A-Sakovich committed:
Add/update localized error messages related to PIN

Anton-A-Sakovich committed:
Add `PinStrengthToErrorMessage` value converter

Anton-A-Sakovich committed:
Update `DefinePinView` and its view model

Add PIN strength validation, corresponding error messages, and password
mask togglers.

Anton-A-Sakovich committed:
Update `EqualsParameter`

Update a summary comment.

Anton-A-Sakovich committed:
Update `DefinePinView.xml`

Some stylistic changes.

Anton-A-Sakovich committed:
Update Define PIN page and view model

Fix the OK button not being disabled when the PIN and the retyped PIN
do not match.

Anton-A-Sakovich committed:
Update `IdApp.xml`

Anton-A-Sakovich committed:
Update `DefinePinViewModel`

Reset `EnteringPinStarted` and `EnteringRetypedPinStarted` properties
when performing `ClearStepState()`.

Anton-A-Sakovich committed:
Update `DefinePinViewModel`

Consider `null` and an empty string equal for the purposes of
`PinsMatch` property.

Anton-A-Sakovich committed:
Update `DefinePinViewModel`

Update a summary comment.

Anton-A-Sakovich committed:
Update `DefinePinViewModel`

Update a summary comment.

ignatiucnetinfo reviewed (commented):
Change the local and parameter variables to the PascalCase

Anton-A-Sakovich committed:
Update `DefinePinView`

Add padding around the password toggler eye glyph in order to increase
the area responding to user taps without increasing the eye visual size.

Anton-A-Sakovich committed:
Update `EqualsParameter`

Make all parameters / locals PascalCase.

Anton-A-Sakovich committed:
Update `PinStrengthToErrorMessage`

Make all parameters / locals PascalCase.

Anton-A-Sakovich committed:
Update `DefinePinViewModel`

Make all parameters / locals PascalCase.

Anton-A-Sakovich committed:
Update `IdApp.xml`
